framework/src/main/kotlin/br/com/zup/beagle/operations/builtin/Operations.kt

```diff
@@ -0,0 +1,140 @@
+
+fun not(param: Bind<Boolean>): Bind.Expression<Boolean> = createOperation("not", arrayOf(param))
+fun or(vararg params: Bind<Boolean>): Bind.Expression<Boolean> = createOperation("or", params)
+
+/** Array **/
+fun <I> contains(array: Bind<Array<I>>, element: Bind<I>): Bind.Expression<Boolean> =
+    createOperation("contains", arrayOf(array, element))
+
+fun <I> insert(array: Bind<Array<I>>, element: Bind<I>, index: Bind<Int>? = null): Bind.Expression<Array<I>> =
+    createOperation("insert", arrayOf(array, element, index))
+
+fun <I> remove(array: Bind<Array<I>>, element: Bind<I>): Bind.Expression<Array<I>> =
+    createOperation("remove", arrayOf(array, element))
+
+fun <I> removeIndex(array: Bind<Array<I>>, index: Bind<Int>): Bind.Expression<Array<I>> =
+    createOperation("removeIndex", arrayOf(array, index))
+
+fun <I> union(firstArray: Bind<Array<I>>, secondArray: Bind<Array<I>>): Bind.Expression<Array<I>> =
+    createOperation("union", arrayOf(firstArray, secondArray))
+
+/** other **/
+fun isEmpty(param: Bind<Boolean>): Bind.Expression<Boolean> = createOperation("isEmpty", arrayOf(param))
+fun isNull(param: Bind<Boolean>): Bind.Expression<Boolean> = createOperation("isNull", arrayOf(param))
+fun length(param: Bind<Array<*>>): Bind.Expression<Int> = createOperation("length", arrayOf(param))
+
+fun <O> createOperation(operationType: String, params: Array<out Any?>): Bind.Expression<O> {
+    val values = params.filterNotNull().map {
+        resolveParam(it as Bind<*>)
+    }
+    return expressionOf("@{${operationType}(${values.joinToString(",")})}")
+}
+
+private fun <I> resolveParam(param: Bind<I>?): Any? {
+    return param?.let {
+        if (param is Bind.Expression && param.value.isNotEmpty()) {
+            param.value.drop(REMOVE_AT_SIGN_AND_OPEN_KEYS).dropLast(REMOVE_KEYS_END)
+        } else {
+            val resultValue = (param as Bind.Value).value
+            if (resultValue is String) {
+                "'${resultValue}'"
+            } else {
+                resultValue
+            }
+        }
+    } ?: run {
+        param
+    }
+}
+
+operator fun Bind<String>.plus(param: Bind<String>): Bind<String> {
+    return if (param is Bind.Expression) {
+        resolvePlusWhenBindIsExpression(param)
+    } else {
+        resolvePlusWhenBindIsValue(param)
+    }
+}
+
+private fun Bind<String>.resolvePlusWhenBindIsExpression(param: Bind.Expression<String>): Bind<String> {
+    return when (this is Bind.Expression) {
+        true -> expressionOf(this.value + param.value)
+        else -> expressionOf((this as Bind.Value).value + param.value)
+    }
+}
+
+private fun Bind<String>.resolvePlusWhenBindIsValue(param: Bind<String>): Bind<String> {
+    return when (this is Bind.Expression) {
+        true -> expressionOf(this.value + (param as Bind.Value).value)
+        else -> expressionOf((this as Bind.Value).value + (param as Bind.Value).value)
+    }
+}
```

Comment:
one way to simplify all this logic is to change the Bind structure and it won't affect anything:

```kotlin

/**
 * Bind is a representation of an expression in the form of a string that can then become any type of data
 */
sealed class Bind<T> : Serializable {

    abstract val value: Any

    /**
     * Represents the expression of bind
     */
    data class Expression<T>(override val value: String) : Bind<T>()

    /**
     * Represents the value of bind
     */
    data class Value<T : Any>(override val value: T) : Bind<T>()

    companion object {

        /**
         * Transform the reference value of the expression string to Bind.Expression<Type>
         */
        fun <T> expression(expression: String) = Expression<T>(expression)

        /**
         * Transform Type value to Bind<Type>.
         */
        fun <T : Any> constant(value: T) = Value(value)

        /**
         * Checks if the value is null. Returns if the value is not null.
         */
        fun <T : Any> constantNullable(value: T?) = value?.let { constant(it) }
    }
}

/**
 * Transform the reference value of the expression string to Bind.Expression<Type>
 */
fun <T> expressionOf(expression: String) = Bind.expression<T>(expression)

/**
 * Transform Type value to Bind<Type>.
 */
fun <T : Any> constant(value: T) = Bind.constant(value)

/**
 * Checks if the value is null. Returns if the value is not null.
 */
fun <T : Any> constantNullable(value: T?) = Bind.constantNullable(value)

```

```suggestion
operator fun Bind<String>.plus(param: Bind<String>): Bind<String> {
    return expressionOf(this.value.toString() + param.value.toString())
}
```